docs/docs/tutorials/noirjs_app.md

```diff
@@ -1,122 +1,101 @@
+keywords: [how to, guide, javascript, typescript, noir, barretenberg, zero-knowledge, proofs, app]
 sidebar_position: 0
 ---
 
-NoirJS works both on the browser and on the server, and works for both ESM and CJS module systems. In this page, we will learn how can we write a simple test and a simple web app to verify the standard Noir example.
+NoirJS is a set of packages meant to work both in a browser and a server environment. In this tutorial, we will build a simple web app using them. From here, you should get an idea on how to proceed with your own Noir projects!
 
 You can find the complete app code for this guide [here](https://github.com/noir-lang/tiny-noirjs-app).
 
-## Before we start
+## Setup
 
 :::note
 
-Feel free to use whatever versions, just keep in mind that Nargo and the NoirJS packages are meant to be in sync. For example, Nargo 0.18.x matches `noir_js@0.18.x`, etc.
+Feel free to use whatever versions, just keep in mind that Nargo and the NoirJS packages are meant to be in sync. For example, Nargo 0.19.x matches `noir_js@0.19.x`, etc.
 
-In this guide, we will be pinned to 0.17.0.
+In this guide, we will be pinned to 0.19.4.
 
 :::
 
-Make sure you have Node installed on your machine by opening a terminal and executing `node --version`. If you don't see a version, you should install [node](https://github.com/nvm-sh/nvm). You can also use `yarn` if you prefer that package manager over npm (which comes with node).
+Before we start, we want to make sure we have Node and Nargo installed.
 
-First of all, follow the the [Nargo guide](../getting_started/installation/index.md) to install nargo version 0.17.0 and create a new project with `nargo new circuit`. Once there, `cd` into the `circuit` folder. You should then be able to compile your circuit into `json` format and see it inside the `target` folder:
+We start by opening a terminal and executing `node --version`. If we don't get an output like `v20.10.0`, that means node is not installed. Let's do that by following the handy [nvm guide](https://github.com/nvm-sh/nvm?tab=readme-ov-file#install--update-script).
 
-```bash
-nargo compile
+As for `Nargo`, we can follow the the [Nargo guide](../getting_started/installation/index.md) to install it. If you're lazy, just run this on a terminal:
+
+```sh
+curl -L https://raw.githubusercontent.com/noir-lang/noirup/main/install | bash
 ```
 
-Your folder structure should look like:
+## Our project
+
+ZK is a powerful technology. An app that doesn't reveal one of the inputs to *anyone* is almost unbelievable, yet Noir makes it as easy as a single line of code.
+
+In fact, it's so simple that it comes nicely packaged in `nargo`. Let's do that!
+
+### Nargo
+
+Run `nargo new circuit`.
+
+And... That's about it. Your program is ready to be compiled and run.
+
+To compile, let's `cd` into the `circuit` folder to enter our project, and call `nargo compile`. This compiles our circuit into `json` format and add it to a new `target` folder.
```

Comment:
Maybe put `nargo compile` onto a new line as well. I like when commands are on their own lines so that its more obvious and it comes with the copy button

```
nargo compile
```